src/main/java/net/dv8tion/jda/bot/sharding/DefaultShardManager.java

```diff
@@ -742,6 +742,15 @@ public void setIdleProvider(IntFunction<Boolean> idleProvider)
         this.idleProvider = idleProvider;
     }
 
+    @Override
+    public void setPresenceProvider(IntFunction<OnlineStatus> statusProvider, IntFunction<Game> gameProvider)
```

Comment:
Use wildcards for objects such as game. `IntFunction<? extends Game>` to allow passing extensions (possibly important later on).